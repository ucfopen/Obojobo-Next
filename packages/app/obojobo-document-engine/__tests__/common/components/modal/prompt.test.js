import React from 'react'
import { mount } from 'enzyme'

import Prompt from 'src/scripts/common/components/modal/prompt'

describe('Prompt Modal', () => {
	beforeEach(() => {
		jest.clearAllMocks()
	})

	test('Prompt component', () => {
		const component = mount(<Prompt />)
		const tree = component.html()

		expect(tree).toMatchSnapshot()
	})

	test('Prompt component calls onConfirm from props', () => {
		const onConfirm = jest.fn()

		const component = mount(<Prompt onConfirm={onConfirm} />)

		component
			.find('button')
			.at(2)
			.simulate('click')

		expect(onConfirm).toHaveBeenCalled()
	})

	test('Prompt component focuses on first element', () => {
		const component = mount(<Prompt content={{}} />)

		component.instance().focusOnFirstElement()

		expect(component.html()).toMatchSnapshot()
	})

	test('Prompt component changes value', () => {
		const onConfirm = jest.fn()

		const component = mount(<Prompt onConfirm={onConfirm} />)

		component
			.find('input')
			.at(1)
			.simulate('change', { target: { value: 'changed' } })

		expect(component.html()).toMatchSnapshot()
	})

	test('Prompt component deals with keydown', () => {
		const onConfirm = jest.fn()

		const component = mount(<Prompt onConfirm={onConfirm} />)

		component
			.find('input')
			.at(1)
			.simulate('keyDown', { key: 'R' })

		expect(onConfirm).not.toHaveBeenCalled()

		component
			.find('input')
			.at(1)
			.simulate('keyDown', { key: 'Enter' })

		expect(onConfirm).toHaveBeenCalled()
	})
})
