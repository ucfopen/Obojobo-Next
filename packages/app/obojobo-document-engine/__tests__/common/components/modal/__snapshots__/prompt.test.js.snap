// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prompt Modal Prompt component 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog \\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"obojobo-draft--components--delete-button\\"><button>Delete</button></div><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"prompt\\"><label for=\\"common--components--modal--prompt--input\\"></label><input type=\\"text\\" id=\\"common--components--modal--prompt--input\\" size=\\"50\\" value=\\"\\"></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\" contenteditable=\\"false\\"><button class=\\"button\\" contenteditable=\\"false\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\" contenteditable=\\"false\\"><button class=\\"button\\" contenteditable=\\"false\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`Prompt Modal Prompt component changes value 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog \\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"obojobo-draft--components--delete-button\\"><button>Delete</button></div><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"prompt\\"><label for=\\"common--components--modal--prompt--input\\"></label><input type=\\"text\\" id=\\"common--components--modal--prompt--input\\" size=\\"50\\" value=\\"changed\\"></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\" contenteditable=\\"false\\"><button class=\\"button\\" contenteditable=\\"false\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\" contenteditable=\\"false\\"><button class=\\"button\\" contenteditable=\\"false\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`Prompt Modal Prompt component focuses on first element 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog \\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"obojobo-draft--components--delete-button\\"><button>Delete</button></div><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"prompt\\"><label for=\\"common--components--modal--prompt--input\\"></label><input type=\\"text\\" id=\\"common--components--modal--prompt--input\\" size=\\"50\\" value=\\"\\"></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\" contenteditable=\\"false\\"><button class=\\"button\\" contenteditable=\\"false\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\" contenteditable=\\"false\\"><button class=\\"button\\" contenteditable=\\"false\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;
